### PR TITLE
[rustdoc] Fix new clippy lints

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Lots of small things that clippy lints about.

r? @fmease

### CODE REVIEW

File: src/librustdoc/html/length_limit.rs [hunk 1]
```diff
@@ -77,23 +77,16 @@ impl HtmlWithLimit {
...
         );
         self.queued_tags.push(tag_name);
     }
 
     /// Close the most recently opened HTML tag.
     pub(super) fn close_tag(&mut self) {
-        match self.unclosed_tags.pop() {
-            // Close the most recently opened tag.
-            Some(tag_name) => write!(self.buf, "</{tag_name}>").unwrap(),
-            // There are valid cases where `close_tag()` is called without
```
Comment (aDotInTheVoid) [inline src/librustdoc/html/length_limit.rs:91]: Was removing this comment deliberate?

File: src/librustdoc/passes/collect_intra_doc_links.rs [hunk 2]
```diff
@@ -1509,15 +1509,11 @@ impl LinkCollector<'_, '_> {
                     type_ns: candidate(TypeNS),
                     value_ns: candidate(ValueNS).and_then(|v_res| {
                         for (res, _) in v_res.iter() {
-                            match res {
-                                // Constructors are picked up in the type namespace.
```
Comment (aDotInTheVoid) [inline src/librustdoc/passes/collect_intra_doc_links.rs:1513]: Ditto.

**Review by aDotInTheVoid — COMMENTED** [review]
r=me with these comment brought back

File: src/librustdoc/html/length_limit.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/length_limit.rs:91]: It wasn't. Seems like a wrong clippy suggestion.

**Comment by GuillaumeGomez** [comment]
Thanks for the review!

@bors r+ rollup

**Pull request merged**